infrastructure-formula/infrastructure/libvirt/domains.sls

```diff
@@ -67,6 +68,55 @@ vm_uuid_map_{{ machine }}:
   - name: /etc/uuidmap
   - text: '{{ machine }}: {{ salt['cmd.run']('uuidgen') }}'
   - unless: 'grep -q {{ machine }} /etc/uuidmap'
+
+{%- if clusterpillar[cluster].get('storage') == 'local' and 'image' in config %}
+{%- if 'root' in config['disks'] %}
+{%- set root_disk = topdir ~ '/disks/' ~ machine ~ '_root.qcow2' %}
+{%- set image = topdir ~ '/os-images/' ~ config['image'] %}
+write_vmdisk_{{ machine }}_root:
+  file.copy:
+    - name: {{ root_disk }}
+    - source: {{ image }}
+    {%- if pillar.get('force_vmdisk', False) %}
+    - force: true
```

Comment:
Looking at the condition again, it might be somewhat dangerous: If `force_vmdisk` is set and is not clearly False (for example it could be set as `force_vmdisk: fales`), then `- force` will be enabled.

Given how much damage this could cause, I'd recommend to make the check more strict - something like "is boolean and is True". Or even require something like `force_vmdisk: destroy_and_reset`

And looking once more - should it really be `pillar.get` (looking for a top-level pillar), or should this check look at a sub-pillar specific to the machine?

(Personally, I'd even tend to drop `force_vmdisk` because it can cause too much harm ;-)

Reply:
> top-level pillar

Mostly because it's easier to pass on the command line for a single `state.apply` than constructing a nested JSON object. However with your concern, that might actually be beneficial. ;-)

Reply:
What about this?

```
{%- if pillar.get('infrastructure:force_vmdisk', False) is sameas true %}
```

Reply:
Amended as discussed.

Reply:
Thanks, the new name makes it very clear what's going on :-)